models/workitemtype_repository_blackbox_test.go

```diff
@@ -0,0 +1,53 @@
+package models_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/almighty/almighty-core/app"
+	"github.com/almighty/almighty-core/models"
+	"github.com/almighty/almighty-core/resource"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCreateLoadWIT(t *testing.T) {
+	doWithTransaction(t, func(ts *models.GormTransactionSupport) {
+		repo := models.NewWorkItemTypeRepository(ts)
+		wit, err := repo.Load(context.Background(), "foo.bar")
+		assert.IsType(t, models.NotFoundError{}, err)
+		assert.Nil(t, wit)
+		wit, err = repo.Create(context.Background(), nil, "foo.bar", map[string]app.FieldDefinition{
+			"foo": app.FieldDefinition{
+				Required: true,
+				Type:     &app.FieldType{Kind: string(models.KindFloat)},
+			},
+		})
+		assert.Nil(t, err)
+		assert.NotNil(t, wit)
+
+		wit3, err := repo.Create(context.Background(), nil, "foo.bar", map[string]app.FieldDefinition{})
+		assert.IsType(t, models.BadParameterError{}, err)
+		assert.Nil(t, wit3)
+
+		wit2, err := repo.Load(context.Background(), "foo.bar")
+		assert.Nil(t, err)
+		assert.NotNil(t, wit2)
+		field := wit2.Fields["foo"]
+		assert.NotNil(t, field)
+		assert.Equal(t, string(models.KindFloat), field.Type.Kind)
+		assert.Equal(t, true, field.Required)
+		assert.Nil(t, field.Type.ComponentType)
+		assert.Nil(t, field.Type.BaseType)
+		assert.Nil(t, field.Type.Values)
+	})
+}
+
+func doWithTransaction(t *testing.T, todo func(ts *models.GormTransactionSupport)) {
+	resource.Require(t, resource.Database)
+	ts := models.NewGormTransactionSupport(db)
+	if err := ts.Begin(); err != nil {
+		panic(err.Error())
+	}
+	defer ts.Rollback()
```

Comment:
I wouldn't actually call it a waste...many unit tests have little to do with what happens in production (for example, because they mock certain systems away). That doesn't mean they are not testing anything. Same here...even while some things may go wrong at commit time a large number of codepaths are tested. 
In the worst case that stuff would go wrong at commit time, you may get false negatives, but you won't get false positives, so if the test fails, you are certain that something is wrong. On the other hand, if the test succeeds, you are not sure it will not fail in production.

But anyway...no quite sure how we would implement idempotent tests while committing changes to the database.